Building plugin

tobrun opened the pull request:
This PR adds a Building plugin, this plugin API matches Traffic plugin API.

![westminster abbey](https://cloud.githubusercontent.com/assets/2151639/25534468/2492c490-2c34-11e7-9468-63b9f79fab3d.png)

tobrun commented:
I finished the initial stab and this PR is ready for review.

Possible API to add:
 - color?
 - opacity?

zugaldia commented:
@tobrun Agreed. I think an API for the three different values you set on https://github.com/mapbox/mapbox-plugins-android/pull/26/files#diff-a1a6e9c8cc422ca880b018bba0a5741cR101 (`minZoom`, `color`, `opacity`) is all we need for now.

tobrun commented:
I have added the API and reworked the code to reflect. This PR is ready for review, though we should maybe wait with merging until v5.1.0 final is released of the Mapbox GL-native dependency?

zugaldia commented:
> This PR is ready for review, though we should maybe wait with merging until v5.1.0 final is released?

👍 Makes sense to me to wait until the underlying API gets stable and published to release 0.1.0. Meanwhile, I suggest we start publishing SNAPSHOTs of the plugin so that interested devs can start playing with it.

tobrun commented:
This PR has been updated with latest SDK release and adds Light integration (just a setter for now)

cammace commented:
> Duplicate import to line 8 - timber.log.Timber. [RedundantImport]

Checkstyle strikes again 😆

cammace commented on plugins/app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/BuildingActivity.java:
```diff
@@ -0,0 +1,178 @@
+  @BindView(R.id.fabBuilding)
+  View fab;
+
+  @BindView(R.id.seekbar_light_radialCoordinate)
```
half resources use camel casing, the rest use snake

cammace commented on plugins/building/gradle.properties:
```diff
@@ -0,0 +1,4 @@
+POM_ARTIFACT_ID=mapbox-android-building-plugin
```
To keep consistency around plugins, the artifact id should be `mapbox-android-plugin-building`

cammace commented on plugins/building/gradle.properties:
```diff
@@ -0,0 +1,4 @@
+POM_ARTIFACT_ID=mapbox-android-building-plugin
+POM_NAME=Mapbox Android Plugins
+POM_DESCRIPTION=Mapbox Android Plugins (Building)
```
Mapbox Android Building Plugin

cammace commented on plugins/building/proguard-rules.pro:
```diff
@@ -0,0 +1,25 @@
+# Add project specific ProGuard rules here.
```
file can be deleted?

cammace commented:
Went ahead and fixed the issues I raised previously. One thing I noticed is that we use `setVisibility(boolean visible)` in this plugin but in traffic we have `toggle()`. Since they essentially do the same thing, we should make the APIs the same (I prefer the `setVisibility` API).

Another addition that should be made in this PR would be the zoom function modifies the building height so they don't just pop in.

cammace commented:
If this is supported across different styles, we should make sure everything is being redrawn correctly when styles change.

zugaldia commented on plugins/build.gradle:
```diff
@@ -10,6 +10,7 @@ buildscript {
 allprojects {
     repositories {
         jcenter()
+        maven { url "http://oss.sonatype.org/content/repositories/snapshots/" }
```
What's this needed for?

cammace replied:
leftover from when we were using map SDK snapshot I think. I removed it.

zugaldia commented on plugins/app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/BuildingActivity.java:
```diff
@@ -0,0 +1,179 @@
+  }
+
+  private void initLightSeekbar() {
+    seekbarRadialCoordinate.setMax(24); // unknown?
```
What does `unknown` mean in this context?

cammace replied:
@tobrun could speak more on this, I'm not sure either.

tobrun replied:
I'm not sure what the maximum radius is for a light source, refs https://en.wikipedia.org/wiki/Spherical_coordinate_system

zugaldia replied:
Could we add a note on code mentioning this?

zugaldia commented on plugins/app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/BuildingActivity.java:
```diff
@@ -0,0 +1,179 @@
+
+    @Override
+    public void onStartTrackingTouch(SeekBar seekBar) {
+
```
Could we add a comment or a Timber call in the unused methods to explicitly note they aren't used on purpose?
